pkg/crd/types.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crd
+
+import (
+	"github.com/kubernetes-incubator/external-dns/endpoint"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DNSZone is the internal represenation of the DNSZone CRD
+type DNSZone struct {
```

Comment:
How does the DNS providers be made aware of this new `DNSZone`. 
DNS providers are the actual piece of code which would create/manage these `DNSZone`s

Reply:
Yep, there is some work to be done on the providers as well. I am on AWS so I can extend and test that one. Potentially the Google one too. Not sure about the others.